### ISSUE TITLE
RFC handling

### ISSUE THREAD

**Issue by rugk** [Langenscheiss/bibitnow#33]
They are already quoted in a quite good way, thanks to metadata I guess.

However taking https://tools.ietf.org/html/rfc7230 as an example, it e.g. misses the ISSN.

This is how it is **currently done**:
```
@online{Reschke2009Jul,
	author = {Reschke, Julian F. and Fielding, Roy T.},
	title = {{Hypertext Transfer Protocol (HTTP/1.1): Message Syntax and Routing}},
	date = {2009-07-27},
	urldate = {2020-07-30},
	language = {english},
	url = {https://tools.ietf.org/html/rfc7230}
}
```

Also, I guess, standards are quoted quite heavily, so it is worth it… :blush: 

Also note, of course :wink:, there is an RFC for how to quote them:
* https://tools.ietf.org/html/draft-carpenter-rfc-citation-recs-01#section-5.2
* https://tex.stackexchange.com/a/161413/98645

---

Using [this generator](https://notesofaprogrammer.blogspot.com/2014/11/bibtex-entries-for-ietf-rfcs-and.html) I get this for the example:
```
@techreport{RFC7230,
  author = {R. Fielding and J. Reschke},
  title = {Hypertext Transfer Protocol (HTTP/1.1): Message Syntax and Routing},
  howpublished = {Internet Requests for Comments},
  type = {RFC},
  number = {7230},
  year = {2014},
  month = {June},
  issn = {2070-1721},
  publisher = {RFC Editor},
  institution = {RFC Editor},
  url = {http://www.rfc-editor.org/rfc/rfc7230.txt},
  note = {\url{http://www.rfc-editor.org/rfc/rfc7230.txt}},
}
```

Note, I personally would like some changes:
* as usual, use `date`, not year+month, but we
* It should really say `Internet Engineering Task Force` somewhere (as in the heading on that RFC actually), IMHO in the `institution` field – no need to duplicate that.
* Of course no need to duplicate the URL in `note`.
* As the Stackoverflow answer says, `shorthand` is useful to add. (maybe possibly also as the note, because this is otherwise not smentioned in the entry then)

So in my case, I ended up with:
```
@techreport{rfc7230,
  author = {R. Fielding and J. Reschke},
  title = {Hypertext Transfer Protocol (HTTP/1.1): Message Syntax and Routing},
  howpublished = {Internet Requests for Comments},
  type = {RFC},
  number = {7230},
  year = {2014},
  month = {June},
  issn = {2070-1721},
  publisher = {RFC Editor},
	institution = {{Internet Engineering Task Force}},
	shorthand = {RFC7230},
	date = {2014-06},
	urldate = {2020-07-30},
	language = {english},
	url = {https://tools.ietf.org/html/rfc7230}
}
```

**Comment by Langenscheiss** [comment]
Hi.
That's a new citation type and a lot of new data fields that websites typically do not provide much information about (e.g. howpublished or institution). But since your request is so far limited to rfcs and since technical reports have to be supported at some point anyhow, I guess I can introduce those fields and leave them blank except for ietf sites. At the moment, I am implementing book chapters which already takes quite a bit of time. But I will put it on the list. I agree quoting standards has some relevance so I should look at it at some point. For the moment, I guess you can use the generator and maybe run the output through a script to get your desired result. Once it is supported, you can of course configure all the fields the way you want as usual.

**Comment by rugk** [comment]
Yeah, maybe you can simplify your work here, because of the standard: https://tools.ietf.org/html/draft-carpenter-rfc-citation-recs-01#section-5.2

It e.g. says you could possibly download a finished bib file online:
> We also RECOMMEND that the RFC Editor create and maintain a canonical
   BibTeX file at a stable public location on the web server "www.rfc-
   editor.org", so that authors using BibTeX can easily obtain a BibTeX
   file with all issued RFCs.

Though [the generator says](https://notesofaprogrammer.blogspot.com/2014/11/bibtex-entries-for-ietf-rfcs-and.html):
> Although the draft recommends that the RFC Editor create and maintain a canonical BibTeX file at a stable public location on the web server "www.rfc-editor.org", the BibTex files do not seem to have appeared on the website.

But maybe you can use the generator's source code or so…

**Comment by Langenscheiss** [comment]
Thanks for the information.
Unfortunately, this all does not help. I cannot simply use another source code, because it needs to be integrated into the whole logic and structure of the extension. The thing is this. Each additional bibfield needs 
- rules to extract the information, possibly site specific, and both statically and dynamically
- rules to parse the information in a reasonable and stable fashion, including character replacements in different latex modes
- rules how the citation output formats -- and not just bibtex but all other formats -- process the parsed information
- rules to be included into the format string system (custom formats, redirection links, etc.)

Of course you can cut corners in the first implementation, like i did with language (which now is also available in formats other than bibtex, and accessible through format strings), but eventually, all this needs to be set up for a well-working system. Now, just for a single source type, this requires me to do all this loop for several bibfields. That's an enormous amount of work, so simply downloading a (possibly corrupt) bibtex file or integrating someone else's source code (which most likely will not fit easily into the existing system and might also produce bad results in some cases) simply won't do it. And that's not even talking about possible permission issues of using source code or of performing cross-site requests that again asks permissions from the user and makes it harder to get it through review at Google/Mozilla.
